### PR TITLE
fix to make setup.cfg work with setuptools v67.0.0

### PR DESCRIPTION
**Pull request by wxtim**
A very small change to the format of the setup.cfg to make it work with the latest version of setuptools.

Non fast-tests cancelled.

See also https://github.com/pypa/setuptools/blob/v67.0.0/CHANGES.rst

(Still digging to find out what part of PEP508 or 440 was the problem)

Caused by [this PR](https://github.com/pypa/packaging/pull/634) fixing.

[Reference: Using setuptools comments to make requirements py version dependent](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html)

### CODE REVIEW

**Comment by MetRonnie** [comment]
> pkg_resources.DistributionNotFound: The 'metomi-isodatetime==1!3.0.*' distribution was not found and is required by cylc-flow

https://github.com/cylc/cylc-flow/actions/runs/4042658912/jobs/6950630795#step:10:358

Seems to be a bug in `setuptools` or `packaging` for Python 3.7?

**Comment by wxtim** [comment]
> > pkg_resources.DistributionNotFound: The 'metomi-isodatetime==1!3.0.*' distribution was not found and is required by cylc-flow
> 
> https://github.com/cylc/cylc-flow/actions/runs/4042658912/jobs/6950630795#step:10:358
> 
> Seems to be a bug in `setuptools` or `packaging` for Python 3.7?

Looks like it :(

File: setup.cfg [hunk 1]
```diff
@@ -67,7 +67,8 @@ install_requires =
     graphene>=2.1,<3
     # Note: can't pin jinja2 any higher than this until we give up on Cylc 7 back-compat
     jinja2==3.0.*
-    metomi-isodatetime==1!3.0.*
+    metomi-isodatetime==1!3.0.*; python_version > "3.7"
+    metomi-isodatetime==1!3; python_version == "3.7"
```
[the remark this answers is not in the document]
Comment (wxtim) [reply setup.cfg]: OK - good point - that works!

File: setup.cfg [hunk 2]
```diff
@@ -72,13 +72,13 @@ install_requires =
     protobuf>=4.21.2,<4.22.0
     psutil>=5.6.0
     pyzmq==22.*
-    setuptools>=49
+    setuptools>=49; python_version > "3.7"
+    setuptools>=49, <67; python_version == "3.7"
```
Comment (MetRonnie) [inline setup.cfg:76]: Ah actually I have just looked at each fast tests' logs and found that for Python 3.8-10, it's using a cached version of setuptools that is lower than 67 anyway, so maybe the problem will still be present for those versions too

Comment (MetRonnie) [reply setup.cfg:76]: Yep, problem present on all versions of Python unfortunately: https://github.com/MetRonnie/cylc-flow/actions/runs/4043433284/jobs/6952341119

Comment (wxtim) [inline setup.cfg:76]:
```suggestion
    setuptools>=49, <67
```

File: setup.cfg [hunk 3]
```diff
@@ -72,13 +72,12 @@ install_requires =
     protobuf>=4.21.2,<4.22.0
     psutil>=5.6.0
     pyzmq==22.*
-    setuptools>=49
+    setuptools>=49, <67
```
Comment (oliver-sanders) [inline setup.cfg:75]: Please apply this to the `conda-environment.yml` file two (they should always match!)

File: setup.cfg [hunk 4]
```diff
@@ -72,13 +72,12 @@ install_requires =
     protobuf>=4.21.2,<4.22.0
     psutil>=5.6.0
     pyzmq==22.*
-    setuptools>=49
+    setuptools>=49, <67; python_version == "3.7"
```
Comment (oliver-sanders) [inline setup.cfg:75]: I think the conclusion was that this issue is generic to all Python versions? @MetRonnie

**Review by oliver-sanders — APPROVED** [review]

File: setup.cfg (hunk 3 above)
Comment (MetRonnie) [inline setup.cfg:75]:
```suggestion
    # https://github.com/pypa/setuptools/issues/3802
    setuptools>=49, <67
```

**Comment by oliver-sanders** [comment]
This PR was raised against master but it should have been "8.1.x", I didn't spot that.

Please go through the check-list with PRs, the last check covers this.

**Pull request merged**